JamfPro/JamfPro/JamfPro.pq

```diff
@@ -271,751 +117,484 @@ UAPIResource = (baseurl as text, relativepath as text, optional query as record,
-        md_group=mdgroup[mobile_devices],
-        result =[mdgroup=md_group]
-    in
-        result;
-
-legacyComputerApplicationsImpl = (ComputerApps as table) as table =>
-    let
-         #"Expanded computerDetails" = Table.ExpandRecordColumn(ComputerApps, "computerDetails", {"computerApplications"}, {"computerDetails.computerApplications"}),
-    #"Expanded computerDetails.computerApplications" = Table.ExpandListColumn(#"Expanded computerDetails", "computerDetails.computerApplications"),
-    #"Expanded computerDetails.computerApplications1" = Table.ExpandRecordColumn(#"Expanded computerDetails.computerApplications", "computerDetails.computerApplications", {"name", "path", "version"}, {"computerDetails.computerApplications.name", "computerDetails.computerApplications.path", "computerDetails.computerApplications.version"})
-in
-    #"Expanded computerDetails.computerApplications1";
-
-computerApplicationsImpl = (baseUrl as text, pages_table as table) as table =>
-    let
         sections = List.Transform({
-            "Applications"
+            "General",
+            "Hardware",
+            "Operating_System",
+            "User_And_Location",
+            "Purchasing",
+            "Disk_Encryption",
+            "Security"
         }, Text.Upper),
 
-        queryString = Text.Combine(List.Transform(sections, Text.Upper), "&section="),
         pageSize = "100",
 
-        temp_table = Table.AddColumn(pages_table, "response", each Table.FromRecords(UAPIResource(baseUrl, "uapi/v1/computers-inventory", [#"page-size" = pageSize, page = Text.From(_[Column1]), section = sections])[results])),
-        computer_table = Table.Combine(temp_table[response]),
-        selected_table = Table.SelectColumns(computer_table, {"id", "applications"}),
-        expand_applications = Table.ExpandListColumn(selected_table, "applications"),
-        filtered_table = Table.SelectRows(expand_applications, each [applications] <> null),
-        expand_applications2 = Table.ExpandRecordColumn(filtered_table, "applications", {"name", "path", "version", "macAppStore", "sizeMegabytes", "bundleId", "updateAvailable", "externalVersionId"})
-    in
-        expand_applications2;
 
-computerConfigurationProfilesImpl = (baseUrl as text, pages_table as table) as table =>
-    let
-        sections = List.Transform({
-            "Configuration_Profiles"
-        }, Text.Upper),
-
-        queryString = Text.Combine(List.Transform(sections, Text.Upper), "&section="),
-        pageSize = "100",
-
-        temp_table = Table.AddColumn(pages_table, "response", each Table.FromRecords(UAPIResource(baseUrl, "uapi/v1/computers-inventory", [#"page-size" = pageSize, page = Text.From(_[Column1]), section = sections])[results])),
+        temp_table = Table.AddColumn(pages_table, "response", each Table.FromRecords(UAPIResource(baseUrl, "uapi/v1/computers-inventory", [#"page-size" = pageSize, page = Text.From(Number.From(_[Column1]) - 1), section = sections])[results])),

```

Comment:
I think we determined the `Web.Contents` function must transform the `sections` list into the appropriate `FOO,BAR,BAZ` format for the URL, but I still think it is weird. 😂